src/libraries/System.Text.Json/tests/Serialization/WriteValueTests.cs

```diff
@@ -285,5 +287,38 @@ static void WriteAndValidate(object input, Type type, string expected, JsonSeria
                 }
             }
         }
+
+        public class CustomClassToExceedMaxBufferSize
+        {
+            private static readonly string s_name = new string('a', 100_000_000);//Large enough value to cause integer overflow exception when allocating buffer and small enought to not cause a "The JSON value of length X is too large and not supported."
+            public string GetName1 => s_name;
+            public string GetName2 => s_name;
+            public string GetName3 => s_name;
+            public string GetName4 => s_name;
+            public string GetName5 => s_name;
+            public string GetName6 => s_name;
+            public string GetName7 => s_name;
+            public string GetName8 => s_name;
+            public string GetName9 => s_name;
+            public string GetName10 => s_name;
+            public string GetName11 => s_name;
+            public string GetName12 => s_name;
+            public string GetName13 => s_name;
+            public string GetName14 => s_name;
+            public string GetName15 => s_name;
+            public string GetName16 => s_name;
+            public string GetName17 => s_name;
+            public string GetName18 => s_name;
+            public string GetName19 => s_name;
+            public string GetName20 => s_name;
+        }
+
+        [ConditionalFact(nameof(IsX64)), OuterLoop]
```

Comment:
Skip running this test on linux.

See other examples (feel free to copy the comment):
https://github.com/dotnet/runtime/blob/4f9ae42d861fcb4be2fcd5d3d55d5f227d30e723/src/libraries/System.Text.Json/tests/Utf8JsonReaderTests.ValueTextEquals.cs#L631-L638